src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -129,6 +149,25 @@ public User getUser(String email) {
     return messages;
   }
 
+ /**
+  * Gets replies of a message, or all messages if parent is null.
+  *
+  * @return a list of any replies to a certain parent message, sorted by time descending. If user is null, returns all replies in the Datastore.
+  */
+  public List<Message> getReplies(String parent) {
```

Comment:
This nearly identical to getAllMessages. I wonder if there's a way to consolidate them without compromising readability? I wouldn't worry about it right now in terms of prioritization, just wanted to mention the possibility.